lib/matplotlib/ticker.py

```diff
@@ -1614,10 +1614,10 @@ def __call__(self):
     def tick_values(self, vmin, vmax):
         if vmax < vmin:
             vmin, vmax = vmax, vmin
-        vmin = self._base.ge(vmin)
         base = self._base.get_base()
-        n = (vmax - vmin + 0.001 * base) // base
-        locs = vmin - base + np.arange(n + 3) * base
+        nmin = round(self._base.ge(vmin) / base)
+        nmax = round(self._base.le(vmax) / base)
+        locs = np.arange(nmin, nmax + 1) * base
```

Comment:
Your proposal can include a value that's above vmax (eg. vmin=.5, vmax=2.5, base=1 will return [1, 2, 3]).  I guess a solution that's safe against floating point errors is
```np.arange(self._base.ge(vmin), self._base.le(vmax) + base / 2, base)```
(the `+ base / 2` should take care of any fpe).